YoCode/Program.cs

```diff
@@ -16,10 +16,9 @@ public static class Program
 
         static void Main(string[] args)
         {
-            
 
             var consoleOutput = new Output(new ConsoleWriter());
-            consoleOutput.PrintIntroduction();
+            var webReport = new Output(new WebWriter());
```

Comment:
Throughout `Main` we have duplicate code where `consoleOutput` and `webReport` are both doing exactly the same thing. It would be better to use the Composite pattern - create a `CompositeWriter` that encapsulates a list of `IPrint` objects and does the same thing to each of them - so all the methods on `CompositeWriter` would be like this:
```
public void Whatever()
{
  foreach(var writer in writers)
  {
    writer.Whatever();
  }
}
```